File: 0242-valid-anagram/0242-valid-anagram.swift
class Solution {
    func isAnagram(_ s: String, _ t: String) -> Bool {
        var mapS : [Character : Int] = [:]
        var mapT : [Character : Int] = [:]
        
        for var char: Character in s{
            mapS[char] = mapS[char, default: 0] + 1
        }

        for var char: Character in t{
            mapT[char] = mapT[char, default: 0] + 1
        }
        if(mapT == mapS){
          return true
        }
        // print(mapS)
        // print(mapT)
        return false
    }
}

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        freq_s = {}
        freq_t = {}
        if (len(s) != len(t)):
            return False
        for char in s:
            if char in freq_s:
                freq_s[char] += 1
            else:
                freq_s[char] = 1
        
        for char in t:
            if char in freq_t:
                freq_t[char] += 1
            else:
                freq_t[char] = 1
                
        for char in freq_s:
            if char not in freq_t:
                return False
            if char in freq_t and freq_s[char] != freq_t[char]:
                return False
            
        return True